src/jit/importer.cpp

```diff
@@ -2584,6 +2584,25 @@ inline IL_OFFSETX Compiler::impCurILOffset(IL_OFFSET offs, bool callInstruction)
     }
 }
 
+//------------------------------------------------------------------------
+// impCanSpillNow: check is it possible to spill all values from eeStack to local variables.
+//
+// Arguments:
+//    prevOpcode - last importer opcode
+//
+// Return Value:
+//    true if it is legal, false if it could be a sequence that we do not want to divide.
+bool Compiler::impCanSpillNow(OPCODE prevOpcode)
+{
+    StackEntry& lastSE = impStackTop(0);
+    GenTreePtr  tree   = lastSE.val;
+    if (prevOpcode == CEE_LDTOKEN)
+    {
+        return false;
```

Comment:
This should have a comment referencing/explaining the InitializeArray sequence that we're avoiding breaking up to guarantee that `impInitializeArrayIntrinsic` can succeed.